api/v1alpha1/imageclusterinstall_types.go

```diff
@@ -34,14 +34,36 @@ const (
 	HostConfigurationSucceededMessage = "Configuration image is attached to the referenced host"
 )
 
-// ClusterConfigSpec defines the desired state of ClusterConfig
-type ClusterConfigSpec struct {
-	clusterinfo.ClusterInfo `json:",inline"`
+// ImageClusterInstallSpec defines the desired state of ImageClusterInstall
+type ImageClusterInstallSpec struct {
+	// ClusterDeploymentRef is a reference to the ClusterDeployment.
+	// +optional
+	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ImageSetRef is a reference to a ClusterImageSet.
+	ImageSetRef hivev1.ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster.
+	// This must be set as soon as all the information is available.
+	// +optional
+	ClusterMetadata *hivev1.ClusterMetadata `json:"clusterMetadata"`
+
+	// Version is the target OCP version for the cluster
+	// TODO: should this use ImageSetRef?
+	Version string `json:"version,omitempty"`
```

Comment:
See what it takes to set this using cluster image set.

Reply:
Remind me why we don't want to get it as input from the user

Reply:
Because it is pretty much duplicated information, right?
It should be part of the cluster image set and the true source of this is the seed image which is already installed.